src/Database/Type/DateTimeType.php

```diff
@@ -105,7 +107,7 @@ public function __construct($name = null)
     /**
      * Convert DateTime instance into strings.
      *
-     * @param string|int|\DateTime|\DateTimeImmutable $value The value to convert.
+     * @param mixed $value The value to convert.
```

Comment:
I am really not a big fan of reverting this. Some effort went into this to make sure we have introspection here on different levels to make sure the "expected" types match. If you, in tests or production, use invalid values to create noops or even exceptions, that is a different thing that we should not use as reason to compromise the good cases for having the expected values annotated here.

Reply:
The type is set as `mixed` for the interface so static analysers are unhappen if you set specific type in the implementation classes. These methods are meant to deal with user input which doesn't have specific type. Also these methods are almost never going to be used directly by end user so types shown by IDEs also isn't really a consideration.